prog/aws/instance.rb

```diff
@@ -181,7 +181,15 @@ def before_run
       raise
     rescue Aws::EC2::Errors::InsufficientInstanceCapacity => e
       if is_runner? && (runner = GithubRunner[vm_id: vm.id])
-        Clog.emit("insufficient instance capacity") { {insufficient_instance_capacity: {vm:, message: e.message}} }
+        families = frame["alternative_families"]
+        next_family = if families && !families.empty? && (index = families.index(vm.family) || -1) < families.length - 1
+          families[index + 1]
```

Comment:
This logic seems more complex than is required. No need to optimize for empty families, and you don't need to check the returned index against the length of families, because if the last index is returned, then `families[index + 1]` will return `nil`:

```suggestion
        next_family = if families
          index = families.index(vm.family) || -1
          families[index + 1]
```